poetry/console/commands/show.py

```diff
@@ -36,11 +37,11 @@ def handle(self):
         if self.option("outdated"):
             self.input.set_option("latest", True)
 
-        locked_repo = self.poetry.locker.locked_repository(True)
+        locked_repo = self.poetry.locker.locked_repository(not self.option("no-dev"))
 
         # Show tree view if requested
         if self.option("tree") and not package:
-            requires = self.poetry.package.requires + self.poetry.package.dev_requires
```

Comment:
I don't think this should be completely removed. We still need to add the dev packages if the `--no-dev` option is not set.

Reply:
You are right. I have corrected the file.